src/Development/IDE/Plugin/Completions.hs

```diff
@@ -22,17 +22,24 @@ import Development.IDE.Core.RuleTypes
 import Development.IDE.Core.Shake
 import Development.IDE.GHC.Util
 import Development.IDE.LSP.Server
-import Development.IDE.Import.DependencyInformation
 
+#ifdef GHC_LIB
+import Data.Maybe
+import Development.IDE.Import.DependencyInformation
+#endif
 
 plugin :: Plugin c
 plugin = Plugin produceCompletions setHandlersCompletion
 
 produceCompletions :: Rules ()
 produceCompletions =
     define $ \ProduceCompletions file -> do
+#ifdef GHC_LIB
         deps <- maybe (TransitiveDependencies [] [] []) fst <$> useWithStale GetDependencies file
         parsedDeps <- mapMaybe (fmap fst) <$> usesWithStale GetParsedModule (transitiveModuleDeps deps)
+#else
+        let parsedDeps = []
+#endif
```

Comment:
This is a bit confusing to read. It just seems like this feature won't work with GHC_LIB?

Reply:
Well, since apparently there is an issue that prevents GHC_LIB from reading Haddocks from interface files, it is not possible to avoid parsing sources when Haddocks/completions are required.